packages/mermaid/src/diagrams/pie/pieRenderer.ts

```diff
@@ -31,50 +31,47 @@ export const draw = (txt, id, _version, diagObj) => {
         ? select(sandboxElement.nodes()[0].contentDocument.body)
         : select('body');
     const doc = securityLevel === 'sandbox' ? sandboxElement.nodes()[0].contentDocument : document;
+    const elem = doc?.getElementById(id);
+    width = elem?.parentElement?.offsetWidth;
```

Comment:
@sidharthv96 I'm having some issues for converting this block of code to the new `info` one. It's related to this issue #4289. It might be better to add a `sandbox` cypress test case for validating it work as expected and increase coverage.

Reply:
the issue is in `document` variable part from d3:

```
const doc = securityLevel === 'sandbox' ? sandboxElement.nodes()[0].contentDocument : document;
```

Reply:
It's appear that sandbox isn't working in all diagrams when adding them to cypress testing or demos. Not sure how it's rendering in `mermaid.live`.

Reply:
I'm a bit confused, why most diagram uses `document` variable from `d3` and set default variables to another values? By doing so, there would be some conflict, i,e, default width in `document` is 984:

https://github.com/mermaid-js/mermaid/blob/3d117815425cbd05105ec0f0192adc0be91931f2/cypress/integration/rendering/pie.spec.js#L54-L57

But the default width value in pie chart is 1200:

https://github.com/mermaid-js/mermaid/blob/3d117815425cbd05105ec0f0192adc0be91931f2/packages/mermaid/src/diagrams/pie/pieRenderer.js#L42-L44